test/unit/helpers/promiseDefer.js

```diff
@@ -0,0 +1,35 @@
+'use strict';
+
+var chai = require('chai');
+var expect = require('chai').expect;
+var PromiseDefer = require('../../../helpers/promiseDefer');
+
+describe('PromiseDefer', function () {
+
+	var promiseDefer;
+	var RESOLVED = 'resolved';
+	var REJECTED = 'rejected';
+
+	beforeEach(function () {
+		promiseDefer = PromiseDefer();
+		promiseDefer.promise.then(function (input) {
+			expect(input.message).to.equal(RESOLVED);
+			input.done();
+		}).catch(function (input) {
+			expect(input.message).to.equal(REJECTED);
+			input.done();
+		});
+	});
+
+	it('resolves', function (done) {
+		expect(promiseDefer.promise.isFulfilled()).to.be.false;
+		promiseDefer.resolve({message: RESOLVED, done: done});
```

Comment:
I would understand what's happening better if the next isFullfilled check would be done on done callback. Not really sure about the code order here.